### ISSUE TITLE
Support TypeScript 3.7

### ISSUE THREAD

**Issue by filipesilva** [angular/tsickle#1096]
While trying to update Angular to `typescript@3.7.0-beta` in https://github.com/angular/angular/pull/32962, I came across this error emitted by `tsickle`:

```
Error: rest parameter does not resolve to a reference type
    at ModuleTypeTranslator.getFunctionTypeJSDoc (/home/circleci/.cache/bazel/_bazel_circleci/9ce5c2144ecf75d11717c0aa41e45a8d/sandbox/processwrapper-sandbox/378/execroot/angular/bazel-out/host/bin/packages/bazel/src/ngc-wrapped/ngc-wrapped.sh.runfiles/npm/node_modules/tsickle/src/module_type_translator.js:424:39)
    at visitFunctionLikeDeclaration (/home/circleci/.cache/bazel/_bazel_circleci/9ce5c2144ecf75d11717c0aa41e45a8d/sandbox/processwrapper-sandbox/378/execroot/angular/bazel-out/host/bin/packages/bazel/src/ngc-wrapped/ngc-wrapped.sh.runfiles/npm/node_modules/tsickle/src/jsdoc_transformer.js:527:75)
    at visitor (/home/circleci/.cache/bazel/_bazel_circleci/9ce5c2144ecf75d11717c0aa41e45a8d/sandbox/processwrapper-sandbox/378/execroot/angular/bazel-out/host/bin/packages/bazel/src/ngc-wrapped/ngc-wrapped.sh.runfiles/npm/node_modules/tsickle/src/jsdoc_transformer.js:913:36)
    at visitNodes (/home/circleci/.cache/bazel/_bazel_circleci/9ce5c2144ecf75d11717c0aa41e45a8d/sandbox/processwrapper-sandbox/378/execroot/angular/bazel-out/host/bin/packages/bazel/src/ngc-wrapped/ngc-wrapped.sh.runfiles/npm/node_modules/typescript/lib/typescript.js:70546:48)
    at visitLexicalEnvironment (/home/circleci/.cache/bazel/_bazel_circleci/9ce5c2144ecf75d11717c0aa41e45a8d/sandbox/processwrapper-sandbox/378/execroot/angular/bazel-out/host/bin/packages/bazel/src/ngc-wrapped/ngc-wrapped.sh.runfiles/npm/node_modules/typescript/lib/typescript.js:70579:22)
...
```

This corresponds to the following code in `tsickle`:

```
                        if (type.flags & ts.TypeFlags.Object &&
                            type.objectFlags & ts.ObjectFlags.Reference) {
                            const typeRef = type;
                            if (!typeRef.typeArguments) {
                                throw new Error('rest parameter does not resolve to a reference type');
                            }
                            type = typeRef.typeArguments[0];
                        }
```

This seems to be another instance of https://github.com/microsoft/TypeScript/issues/33693, a breaking change that affects Angular. It's easy to work around though:

before:
```
          const typeReference = type as ts.TypeReference;
          if (typeReference.typeArguments && typeReference.typeArguments.length === 1) {
            return typeReference.typeArguments[0].symbol;
          }
```
after
```
          const typeReference = type as ts.TypeReference;
          const typeArguments = this.checker.getTypeArguments(typeReference);
          if (typeArguments.length === 1) {
            return typeArguments[0].symbol;
          }
```

**Comment by brad4d** [comment]
Created Google internal issue http://b/142268984

**Comment by chadbr** [comment]
Looking forward to 3.7 support! ❤️

**Comment by filipesilva** [comment]
This seems to be addressed in the `ts37` branch (https://github.com/angular/tsickle/commit/a1e7cce0a0a5f410505e89843f387f98efbe493e) and released as `tsickle@0.38.0`.

**Comment by mprobst** [comment]
That's correct @filipesilva
